src/query/procedure/mg_procedure_impl.hpp

```diff
@@ -675,8 +675,27 @@ struct mgp_vertices_iterator {
   /// @throw anything VerticesIterable may throw
   mgp_vertices_iterator(mgp_graph *graph, memgraph::utils::MemoryResource *memory)
       : memory(memory), graph(graph), vertices(graph->impl->Vertices(graph->view)), current_it(vertices.begin()) {
-    if (current_it != vertices.end()) {
+    memgraph::utils::OnScopeExit clean_up([this] { this->current_v = std::nullopt; });
+    auto *checker = graph->ctx->fine_grained_access_checker;
```

Comment:
```suggestion
    const auto *checker = graph->ctx->fine_grained_access_checker;
```

Furthermore I think this is very similar to `mgp_vertices_iterator_next`, it might make sense to extract the logic into a common function.

Reply:
`graph->ctx->fine_grained_access_checker` can be `nullptr`, so a check is necessary. One more reason to extract this logic into a common function.